orchestrator/graphql/schemas/subscription.py

```diff
@@ -0,0 +1,159 @@
+
+        def to_product_block(product_block: dict[str, Any]) -> SubscriptionProductBlock:
+            def is_resource_type(candidate: Any) -> bool:
+                return isinstance(candidate, (bool, str, int, float, type(None)))
+
+            def requested_resource_type(key: str) -> bool:
+                return not resource_types or key in resource_types
+
+            def included(key: str, value: Any) -> bool:
+                return is_resource_type(value) and requested_resource_type(key) and key not in ("id", "parent")
+
+            return SubscriptionProductBlock(
+                id=product_block["id"],
+                parent=product_block.get("parent"),
+                owner_subscription_id=product_block["owner_subscription_id"],
+                resource_types={to_camel(k): v for k, v in product_block.items() if included(k, v)},
+            )
+
+        product_blocks = (
+            to_product_block(product_block) for product_block in get_all_product_blocks(subscription, tags)
+        )
+        return [product_block for product_block in product_blocks if product_block.resource_types]
+
+    @authenticated_field(description="Returns list of processes of the subscription")  # type: ignore
+    async def processes(
+        self,
+        info: CustomInfo,
+        filter_by: Union[list[GraphqlFilter], None] = None,
+        sort_by: Union[list[GraphqlSort], None] = None,
+        first: int = 10,
+        after: int = 0,
+    ) -> Connection[ProcessType]:
+        filter_by = filter_by or [] + [GraphqlFilter(field="subscriptionId", value=str(self.subscription_id))]
+        return await resolve_processes(info, filter_by, sort_by, first, after)
+
+    @authenticated_field(description="Returns list of subscriptions that use this subscription")  # type: ignore
+    async def in_use_by_subscriptions(
+        self,
+        info: CustomInfo,
+        filter_by: Union[list[GraphqlFilter], None] = None,
+        sort_by: Union[list[GraphqlSort], None] = None,
+        first: int = 10,
+        after: int = 0,
+    ) -> Connection[Annotated["SubscriptionType", strawberry.lazy(".subscription")]]:
+        from orchestrator.graphql.resolvers.subscription import resolve_subscription
+        from orchestrator.services.subscriptions import query_in_use_by_subscriptions
+
+        in_use_by_query = query_in_use_by_subscriptions(self.subscription_id)
+        query_results = in_use_by_query.with_entities(SubscriptionTable.subscription_id).all()
+        subscription_ids = ",".join([str(s.subscription_id) for s in query_results])
+        filter_by = (filter_by or []) + [GraphqlFilter(field="subscriptionIds", value=subscription_ids)]
+        return await resolve_subscription(info, filter_by, sort_by, first, after)
+
+    @authenticated_field(description="Returns list of subscriptions that this subscription depends on")  # type: ignore
+    async def depends_on_subscriptions(
```

Comment:
Lots of similar code to in_use_by_subscriptions

Reply:
I can't see a improvement of splitting stuff without making it even less readable.